sentry_sdk/profiler.py

```diff
@@ -918,7 +918,13 @@ def ensure_running(self):
             # can keep the application running after other threads
             # have exited
             self.thread = threading.Thread(name=self.name, target=self.run, daemon=True)
-            self.thread.start()
+            try:
+                self.thread.start()
+            except RuntimeError:
+                # Unfortunately at this point the interpreter is in a state that no
+                # longer allows us to spawn a thread and we have to bail.
+                self.running = False
+                return
```

Comment:
I find it slightly confusing here that the method is called `ensure_running`, but it is possible for the method to terminate without the thread running. We should have some way to indicate failure, perhaps by raising a custom exception when `ensure_running` cannot start the scheduler or by returning a value to indicate success/failure. We might instead or additionally consider renaming the function to indicate that we may fail to ensure that the scheduler is running and/or add a documentation comment.

If we make this change to indicate when the `ensure_running` fails, we should also make sure to handle failure appropriately in the calling code.